src/integration_tests/rpc_multiple_remote_addrs/main.cc

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) 2016 Alexander Gallego. All rights reserved.
+//
+// std
+#include <chrono>
+#include <iostream>
+// linux
+#include <arpa/inet.h>
+#include <ifaddrs.h>
+#include <net/if.h>
+// seastar
+#include <core/app-template.hh>
+#include <core/distributed.hh>
+#include <core/sleep.hh>
+#include <net/api.hh>
+// smf
+#include "integration_tests/non_root_port.h"
+#include "smf/log.h"
+#include "smf/random.h"
+#include "smf/rpc_generated.h"
+#include "smf/rpc_handle_router.h"
+#include "smf/rpc_recv_context.h"
+#include "smf/rpc_server.h"
+// templates
+#include "integration_tests/demo_service.smf.fb.h"
+
+struct client_info {
+    explicit client_info(seastar::ipv4_addr ip): client(smf_gen::demo::SmfStorageClient::make_shared(ip)), ip(ip.ip) {};
+    client_info(): client(nullptr), ip(0), ip_str("") {};
+    seastar::shared_ptr<smf_gen::demo::SmfStorageClient> client = nullptr;
+    uint32_t ip = 0;
+    char ip_str[INET_ADDRSTRLEN]{};
+};
+
+class storage_service : public smf_gen::demo::SmfStorage {
+  virtual seastar::future<smf::rpc_typed_envelope<smf_gen::demo::Response>>
+  Get(smf::rpc_recv_typed_context<smf_gen::demo::Request> &&rec) final {
+    char address_str[INET_ADDRSTRLEN] = {};
+    smf::rpc_typed_envelope<smf_gen::demo::Response> data;
+    if (rec) {
+        DLOG_DEBUG("Got request name {}", rec->name()->str());
+        inet_ntop(AF_INET, &rec.ctx->remote_address.as_posix_sockaddr_in().sin_addr, address_str, INET_ADDRSTRLEN);
+        data.data->name = seastar::to_sstring(address_str);
+    }
+    data.envelope.set_status(200);
+    return seastar::make_ready_future<
+        smf::rpc_typed_envelope<smf_gen::demo::Response>>(std::move(data));
+  }
+};
+
+#define MAX_CLIENTS 2
+
+struct ifaddrs* next_valid_interface(struct ifaddrs* ifa) {
```

Comment:
just a thought, i wonder if this `next()` can be handle by just a `std::vector` instead, since you have iterators do this for you 
```
std::vector<struct ifaddrs> v = {...}; 
std::for_each( v.begin(), v.end(), [ ] (auto &next) {
              .... do something w/ next;
});
```